Using a Spack built GCC the build still errors out:
```
10 errors found in build log:
     1896    [ 58%] Building CXX object Source/CMakeFiles/CMakeLib.dir/LexerParser/cmDependsJavaParser.cxx.o
     1897    [ 58%] Building CXX object Source/CMakeFiles/CMakeLib.dir/LexerParser/cmCommandArgumentParser.cxx.o
     1898    [ 58%] Building CXX object Source/CMakeFiles/CMakeLib.dir/LexerParser/cmFortranLexer.cxx.o
     1899    [ 58%] Building CXX object Source/CMakeFiles/CMakeLib.dir/LexerParser/cmExprParser.cxx.o
     1900    In file included from /Users/culpo/PycharmProjects/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.0/gcc-9.2.0-ie4mp5d7u3ygeu3icgelktq437acyzw6/include/c
             ++/9.2.0/stdlib.h:36,
     1901                     from /private/var/folders/gy/g0z3m6dn0nz53nsttz2q0flm0000gn/T/culpo/spack-stage/spack-stage-cmake-3.18.2-5ylrs2biuqeuiwpjd4fm2osnhdqd45vm/spack-
             src/Source/LexerParser/cmFortranLexer.cxx:232:
  >> 1902    /Users/culpo/PycharmProjects/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.0/gcc-9.2.0-ie4mp5d7u3ygeu3icgelktq437acyzw6/include/c++/9.2.0/cstdlib:132:1
             1: error: '::aligned_alloc' has not been declared
     1903      132 |   using ::aligned_alloc;
     1904          |           ^~~~~~~~~~~~~
  >> 1905    make[2]: *** [Source/CMakeFiles/CMakeLib.dir/LexerParser/cmFortranLexer.cxx.o] Error 1
     1906    make[2]: *** Waiting for unfinished jobs....
     1907    In file included from /Users/culpo/PycharmProjects/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.0/gcc-9.2.0-ie4mp5d7u3ygeu3icgelktq437acyzw6/include/c
             ++/9.2.0/stdlib.h:36,
     1908                     from /private/var/folders/gy/g0z3m6dn0nz53nsttz2q0flm0000gn/T/culpo/spack-stage/spack-stage-cmake-3.18.2-5ylrs2biuqeuiwpjd4fm2osnhdqd45vm/spack-
             src/Source/LexerParser/cmExprLexer.cxx:232:
  >> 1909    /Users/culpo/PycharmProjects/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.0/gcc-9.2.0-ie4mp5d7u3ygeu3icgelktq437acyzw6/include/c++/9.2.0/cstdlib:132:1
             1: error: '::aligned_alloc' has not been declared
     1910      132 |   using ::aligned_alloc;
     1911          |           ^~~~~~~~~~~~~
     1912    In file included from /Users/culpo/PycharmProjects/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.0/gcc-9.2.0-ie4mp5d7u3ygeu3icgelktq437acyzw6/include/c
             ++/9.2.0/stdlib.h:36,
     1913                     from /private/var/folders/gy/g0z3m6dn0nz53nsttz2q0flm0000gn/T/culpo/spack-stage/spack-stage-cmake-3.18.2-5ylrs2biuqeuiwpjd4fm2osnhdqd45vm/spack-
             src/Source/LexerParser/cmCommandArgumentLexer.cxx:232:
  >> 1914    /Users/culpo/PycharmProjects/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.0/gcc-9.2.0-ie4mp5d7u3ygeu3icgelktq437acyzw6/include/c++/9.2.0/cstdlib:132:1
             1: error: '::aligned_alloc' has not been declared
     1915      132 |   using ::aligned_alloc;
     1916          |           ^~~~~~~~~~~~~
     1917    In file included from /Users/culpo/PycharmProjects/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.0/gcc-9.2.0-ie4mp5d7u3ygeu3icgelktq437acyzw6/include/c
             ++/9.2.0/stdlib.h:36,
     1918                     from /private/var/folders/gy/g0z3m6dn0nz53nsttz2q0flm0000gn/T/culpo/spack-stage/spack-stage-cmake-3.18.2-5ylrs2biuqeuiwpjd4fm2osnhdqd45vm/spack-
             src/Source/LexerParser/cmDependsJavaLexer.cxx:232:
  >> 1919    /Users/culpo/PycharmProjects/spack/opt/spack/darwin-catalina-x86_64/apple-clang-11.0.0/gcc-9.2.0-ie4mp5d7u3ygeu3icgelktq437acyzw6/include/c++/9.2.0/cstdlib:132:1
             1: error: '::aligned_alloc' has not been declared
     1920      132 |   using ::aligned_alloc;
     1921          |           ^~~~~~~~~~~~~
  >> 1922    make[2]: *** [Source/CMakeFiles/CMakeLib.dir/LexerParser/cmExprLexer.cxx.o] Error 1
  >> 1923    make[2]: *** [Source/CMakeFiles/CMakeLib.dir/LexerParser/cmCommandArgumentLexer.cxx.o] Error 1
  >> 1924    make[2]: *** [Source/CMakeFiles/CMakeLib.dir/LexerParser/cmDependsJavaLexer.cxx.o] Error 1
  >> 1925    make[1]: *** [Source/CMakeFiles/CMakeLib.dir/all] Error 2
  >> 1926    make: *** [all] Error 2

See build log for details:
  /var/folders/gy/g0z3m6dn0nz53nsttz2q0flm0000gn/T/culpo/spack-stage/spack-stage-cmake-3.18.2-5ylrs2biuqeuiwpjd4fm2osnhdqd45vm/spack-build-out.txt
```
even though the error seem a lot more like [this](https://stackoverflow.com/a/52777686/771663) than previous Objective C issue.

--- a/var/spack/repos/builtin/packages/cmake/package.py
+++ b/var/spack/repos/builtin/packages/cmake/package.py
@@ -100,14 +100,11 @@ class Cmake(Package):
     variant('openssl', default=True,  description="Enables CMake's OpenSSL features")
     variant('ncurses', default=True,  description='Enables the build of the ncurses gui')
 
-    # Tries to build an Objective-C file from libuv with GCC's C frontend
-    # https://github.com/libuv/libuv/issues/2805
-    # https://gitlab.kitware.com/cmake/cmake/-/merge_requests/4687
-    _msg_gcc_macOS = 'CMake<3.18.0 does not compile with GCC on macOS, ' \
-                     'use %apple-clang.'
-    conflicts('@:3.17.99 %gcc +ownlibs platform=darwin', msg=_msg_gcc_macOS)
-    conflicts('%gcc ~ownlibs ^libuv@:1.37.99 platform=darwin',
-              msg=_msg_gcc_macOS)
+    # Does not compile and is not covered in upstream CI (yet).
+    conflicts('%gcc platform=darwin',
+              msg='CMake does not compile with GCC on macOS yet, '
+                  'please use %apple-clang. '
+                  'See: https://gitlab.kitware.com/cmake/cmake/-/issues/21135')
 
     # Really this should conflict since it's enabling or disabling openssl for
     # CMake's internal copy of curl.  Ideally we'd want a way to have the

